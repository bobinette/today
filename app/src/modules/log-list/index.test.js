import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import { LogList } from '.';

test('render the log list', () => {
  const fetchLogs = jest.fn();
  const onSearchChange = jest.fn();
  const props = {
    logs: fromJS([
      { uuid: '1', content: 'content 1' },
      { uuid: '2', content: 'content 2' },
      { uuid: '3', content: 'content 3' },
    ]),
    q: 'q',
    actions: {
      fetchLogs,
      onSearchChange,
    },
  };

  const tree = shallow(<LogList {...props} />);
  expect(tree).toMatchSnapshot();
  expect(fetchLogs).toHaveBeenCalled();
});
