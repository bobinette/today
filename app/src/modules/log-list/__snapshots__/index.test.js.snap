// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the log list 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogList
    actions={
      Object {
        "fetchLogs": [MockFunction] {
          "calls": Array [
            Array [],
          ],
        },
        "onSearchChange": [MockFunction],
      }
    }
    logs={
      Immutable.List [
        Immutable.Map {
          "uuid": "1",
          "content": "content 1",
        },
        Immutable.Map {
          "uuid": "2",
          "content": "content 2",
        },
        Immutable.Map {
          "uuid": "3",
          "content": "content 3",
        },
      ]
    }
    q="q"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(NewLogInput) />,
        <SearchBar
          className="LogList__searchBar col-md-8 offset-md-2"
          onChange={[MockFunction]}
          q="q"
        />,
        Immutable.List [
          <div>
            <Unknown
              log={
                Immutable.Map {
                  "uuid": "1",
                  "content": "content 1",
                }
              }
            />
          </div>,
          <div>
            <Unknown
              log={
                Immutable.Map {
                  "uuid": "2",
                  "content": "content 2",
                }
              }
            />
          </div>,
          <div>
            <Unknown
              log={
                Immutable.Map {
                  "uuid": "3",
                  "content": "content 3",
                }
              }
            />
          </div>,
        ],
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "LogList__searchBar col-md-8 offset-md-2",
          "onChange": [MockFunction],
          "q": "q",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            log={
              Immutable.Map {
                "uuid": "1",
                "content": "content 1",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "log": Immutable.Map {
              "uuid": "1",
              "content": "content 1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            log={
              Immutable.Map {
                "uuid": "2",
                "content": "content 2",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "log": Immutable.Map {
              "uuid": "2",
              "content": "content 2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            log={
              Immutable.Map {
                "uuid": "3",
                "content": "content 3",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "log": Immutable.Map {
              "uuid": "3",
              "content": "content 3",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(NewLogInput) />,
          <SearchBar
            className="LogList__searchBar col-md-8 offset-md-2"
            onChange={[MockFunction]}
            q="q"
          />,
          Immutable.List [
            <div>
              <Unknown
                log={
                  Immutable.Map {
                    "uuid": "1",
                    "content": "content 1",
                  }
                }
              />
            </div>,
            <div>
              <Unknown
                log={
                  Immutable.Map {
                    "uuid": "2",
                    "content": "content 2",
                  }
                }
              />
            </div>,
            <div>
              <Unknown
                log={
                  Immutable.Map {
                    "uuid": "3",
                    "content": "content 3",
                  }
                }
              />
            </div>,
          ],
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "LogList__searchBar col-md-8 offset-md-2",
            "onChange": [MockFunction],
            "q": "q",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              log={
                Immutable.Map {
                  "uuid": "1",
                  "content": "content 1",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "log": Immutable.Map {
                "uuid": "1",
                "content": "content 1",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              log={
                Immutable.Map {
                  "uuid": "2",
                  "content": "content 2",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "log": Immutable.Map {
                "uuid": "2",
                "content": "content 2",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              log={
                Immutable.Map {
                  "uuid": "3",
                  "content": "content 3",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "log": Immutable.Map {
                "uuid": "3",
                "content": "content 3",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
